### DISCUSSION TITLE
USB upload methods are impure, looking for pure direct Arduino upload methods.

### DISCUSSION THREAD

**Discussion by XanCraft21** [stm32duino/Arduino_Core_STM32#1707] [General]
Hello, the problem i am having with this software is the lack of safe USB upload support. I have discovered that my computer will not allow me to download the required software “STM32 CUBE PROGRAMMER” as it thinks there is a virus hidden inside. I do not believe this is true, but i want to be safe as i am very paranoid about damaging my pc.

The solution i was thinking of for this is to add the ability to directly upload to the boards strait from Arduino via Serial upload without the need to download the cube programmer, as i feel like this software should be an optional choice for extended debugging resources unless they are not needed. I also feel like if there are other users who are as paranoid as i am, even if we don’t feel safe using the cube programmer, we can still have full access to the libraries and an easier ride for first time users.

There is really no alternative in my opinion, other than what i just described above.

It may be possible that no-one cares about my request or concerns, but i would really love to use this software to its full capacity within only one program without the need for any extra drivers or software that my computer won’t allow me to download. I don’t mind using this software for large requirements, but i just need an easier setup for simplicity and safety. Thank you to anyone who is able to reach out to me about my concerns and questions and who can work with the developers to make this possible.

**Comment by fpistm** [comment]
Hi,
this discussion had already been done.
Anyway the STM32CubeProgrammer is not only for debugging resources.
This is the official software for all available upload methods provided by the manufacturer.
Anyway you are right, it is not mandatory as other upload method exist: dedicated bootloader, Mass storage copy (for ST manufactured board like Nucleo, disco,...), dfu utils,...
One other way for end user is to define his own upload method using https://arduino.github.io/arduino-cli/0.21/platform-specification/#platformlocaltxt and https://arduino.github.io/arduino-cli/0.21/platform-specification/#boardslocaltxt

**Reply by XanCraft21** [reply]
Sorry, i wasn’t able to find that this discussion has already been done before. Which upload method allows Serial programming upload directly from arduino without using the cube IDE? I do not need the debugging features at this time, and i just want to save hard drive space and make everything simpler for myself if possible.

**Comment by fpistm** [comment]
As stated usage of STM32CubeProgrammer it is not link to debugging.
You can try this anyway we do not support this, it is custom user and not reliable for maintenance.
https://github.com/stm32duino/Arduino_Tools/issues/39#issuecomment-638006622

**Reply by XanCraft21** [reply]
Ok, i will see what i can do. Even though it’s probably inefficient, i’ll take anything. Thank you for the help.
